### PR TITLE
dep: update parser, dumpling, tidb

### PR DESCRIPTION
**Pull request by lance6716**
### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by leoppro** [comment]
LGTM

**Review by lichunzhu — COMMENTED** [review]
We need to add dumpling prometheus metrics and do a test.

File: dumpling/metrics.go [hunk 1]
```diff
@@ -30,11 +33,19 @@ var (
...
+	export.InitMetricsVector(prometheus.Labels{"task": "", "source_id": ""})
+	export.RegisterMetrics(registry)
 }
 
-func (m *Dumpling) removeLabelValuesWithTaskInMetrics(task string) {
-	dumplingExitWithErrorCounter.DeleteAllAboutLabels(prometheus.Labels{"task": task})
+func (m *Dumpling) removeLabelValuesWithTaskInMetrics(task, source string) {
+	labels := prometheus.Labels{"task": task, "source_id": source}
+	dumplingExitWithErrorCounter.DeleteAllAboutLabels(labels)
+	failpoint.Inject("SkipRemovingDumplingMetrics", func(_ failpoint.Value) {
```
Comment (lichunzhu) [inline dumpling/metrics.go:46]: Why do we need this here?

Comment (lance6716) [reply dumpling/metrics.go:46]: added a failpoint, otherwise integration test will remove all metrics so we can't check

**Review by lichunzhu — COMMENTED** [review]
/lgtm

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**